src/JsonFileHelper.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\Component;
+
+use Symfony\Component\Filesystem\Exception\FileNotFoundException;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+
+class JsonFileHelper
+{
+    public static function read(string $filePath): array
+    {
+        if (!file_exists($filePath)) {
+            throw new FileNotFoundException(null, 0, null, $filePath);
+        }
+
+        $jsonEncoder = new JsonEncoder();
+        $jsonContents = file_get_contents($filePath);
+        return $jsonEncoder->decode($jsonContents, JsonEncoder::FORMAT);
+    }
+
+    public static function write(string $filePath, array $data, bool $formatted = true): void
+    {
+        $filePathDir = pathinfo($filePath, PATHINFO_DIRNAME);
+        if (!is_dir($filePathDir)) {
+            mkdir($filePathDir, 0777, true);
+        }
+
+        $context = [];
+        if ($formatted) {
+            $context = ['json_encode_options' => JSON_PRETTY_PRINT];
+        }
+
+        $jsonEncoder = new JsonEncoder();
+        file_put_contents(
```

Comment:
tady by to melo vyhodit vyjimku pokud je navratova hodnota === false, jinak to nemuselo zapsat

Reply:
Vis za jakych podminek to vraci `false`? Nenapada me jak to otestovat.

Reply:
nebo adresar, co nema write permissions

Reply:
To jsem zkousel, ale kdyz nemas write permission, tak uz sam file_put_contents vyhodi exception. Zatim to nechavam bez testu.

Reply:
to je divny file_put_contents vyjimky nevyhazuje, co to vyhodi za vyjimku? neni to PHPUnitovy convertovany warning?

Reply:
nj, tak to je jasny https://github.com/keboola/php-component/blob/master/src/BaseComponent.php#L65 :)

Reply:
Hmm.. Opravdu se to settuje (tim myslim, jestli to zustane nasettovano z predchozich testu) kdyz tu funkci volam staticky? 

I kdyz spustim samostatne ten jeden test, tak to skonci stejne.

Reply:
Když bys nastavil `processIsolation` tak by to nemělo být nastavené https://github.com/keboola/php-component/blob/ccf0a216c0184d27b4f1db4d7d8042177ef6842a/phpunit.xml.dist#L8

Reply:
no ja nevim presne kterej test myslis, ale tohle se vola v ctoru BaseComponent
jinymi slovy v ramci php-component to takhle bude fungovat vicmene vzdycky, pokud se jsonfilehelper vytahne do vlastni lib, tak to fungovat nemusi

Reply:
aha blbost, https://github.com/keboola/php-component/blob/ccf0a216c0184d27b4f1db4d7d8042177ef6842a/tests/bootstrap.php#L7

Reply:
Poustim test (jen jeden), ktery vubec BaseComponent nevytvari, ciste jen vola `JsonHelper::write()` nic vic a stejne to vyhodi exception.

Reply:
```
public function testWriteToDevFull(): void
    {
        $filePath = '/dev/full';
        $array = ['key'];

        JsonFileHelper::write($filePath, $array);
    }
```

Reply:
protože se to nastavuje i v tom bootstrapu testů.

Reply:
jop https://github.com/keboola/php-component/blob/master/tests/bootstrap.php#L7

Reply:
Jasne, to je ono. Jinak je to warning. Nicmene i tak nevim, jak otestovat, kdy file_put_content vraci false 😕

Reply:
v php-component nijak, tady to vyhazuje vyjimku, takze tady staci otestovat, ze to vyhazuje vyjimku, az se to bude davat do utils, tak tam je to potreba poresit - bud registraci handleru, nebo `@` a otestovani false

Reply:
Ok, tak tu jsem pridal ten test.